app/models/merchant.rb

```diff
@@ -14,6 +14,13 @@ def revenue(date)
       .sum('unit_price_in_cents * quantity')
   end
 
+  def self.most_revenue(number_of_merchants)
+    joins(:invoice_items)
+      .merge(InvoiceItem.successful)
+      .group("merchants.id")
```

Comment:
you can just do `.group(:id)` here if you wanted since you are already within the merchant model.